lib/runners/processor/metrics_fileinfo.rb

```diff
@@ -91,12 +101,59 @@ def last_committed_at
     def analyze_last_committed_at(targets)
       trace_writer.message "Analyzing last commit time..." do
         Parallel.each(targets, in_threads: 8) do |target|
-          stdout, _ = capture3!("git", "log", "-1", "--format=format:%aI", "--", target, trace_stdout: false, trace_command_line: false)
+          stdout, _ = git("log", "-1", "--format=format:%aI", "--", target)
           last_committed_at[target] = stdout
         end
       end
     end
 
+    def code_churn
+      @code_churn ||= {}
+    end
+
+    def analyze_code_churn
+      trace_writer.message "Analyzing code churn..." do
+        count_by_num, commits_by_num = commits_within("-n", "100")
+        days_ago = (Time.parse(commits_by_num[:latest][:datetime]) - 90 * 60 * 60 * 24).iso8601
+        count_by_time, commits_by_time = commits_within("--since", days_ago)
+        outlive_commits = count_by_num > count_by_time ? commits_by_num : commits_by_time
+
+        stdout, _ = git("log", "--reverse", "--format=format:%cI", "--numstat", "#{outlive_commits[:oldest][:sha]}..HEAD")
+        lines = stdout.lines(chomp: true)
+        number_of_commits = 0
+        lines.reject(&:empty?).each do |line|
+          adds, dels, fname = line.split("\t")
+          if adds && dels && fname
+            fname = Pathname(fname)
+            code_churn[fname] = calc_churn(code_churn[fname], adds, dels)
+          else
+            number_of_commits += 1
+          end
+        end
+
+        number_of_commits
```

Comment:
We don't need to count `number_of_commit` one by one because we can use blank line between each commit.
How about this?
```suggestion
        stdout, _ = git("log", "--reverse", "--format=format:", "--numstat", "#{outlive_commits[:oldest][:sha]}..HEAD")
        lines = stdout.lines(chomp: true)
        num_of_commits = lines.empty? ? 0 : lines.count{|l| l.empty?} + 1

        lines.each do |line|
          adds, dels, fname = line.split("\t")
          if adds && dels && fname
            fname = Pathname(fname)
            code_churn[fname] = calc_churn(code_churn[fname], adds, dels)
          end
        end

        num_of_commits
```
NOTE: `git log` outputs as below with `--format=format:` option.
```
~/tmp/runners/test/smokes/metrics_fileinfo/multi_commit$ GIT_DIR=gitmetadata git log --reverse --format="format:" --numstat HEAD^^..HEAD
1       0       hello.rb

1       0       hello.rb
~/tmp/runners/test/smokes/metrics_fileinfo/multi_commit$
```

Reply:
@gnakagaw 
Finding an empty line seems to be far from true intention of counting the number of commits.
How about finding a marker like "#" by adding it with `--format=format:#` option?

Reply:
> How about finding a marker like "#" by adding it with --format=format:# option?
It make the code intention more clearly. 👍

Reply:
Fixed. commit f2330fec10093623231ac97c9539f1ba4a74ec7e